### ISSUE TITLE
Wrangling/Computation: SQL vs. Pandas vs. One-Hot Encoding

### ISSUE THREAD

**Issue by Skenderbeu1405** [georgetown-analytics/Loan-Canoe#18]
**ii. SQL vs. Pandas vs. One Hot Encoding**:

*MC:*
i.  In general, and as already discussed with Ben, we should feel empowered to use SQL as much as possible, in fact, for wrangling and even computation to make ready for the modeling step;

ii. SQL has many advantages over pandas (note: we should use both pandas and SQL, also, consider using psql.with_open ('sql file path') instead of writing sql in notebook cell if the sql is many lines long; also, see my script in 'models' that deploys postgreSQL for wrangling and has links to Documentation). 

**iii. Also, very important,** whenever possible, strong preference for one-hot encoding if leveraging scikit-learn because of the importance of pipelining -- as confirmed with Ben: put everything as much as possible in Pipeline.

**Comment by Anne-Kl** [comment]
Useful link: https://www.datacamp.com/community/tutorials/categorical-data
Shows creation of dummy variables for a data set that isn't all categorical fields.

**Comment by Anne-Kl** [comment]
ALSO: for point (iii), these notebooks rock: https://github.com/georgetown-analytics/machine-learning/tree/master/examples/bbengfort/census

The data is census data with a mix of int/categorical features.  Looking at these tonight and should hopefully have updated code by tomorrow.

**Comment by Anne-Kl** [comment]
Update as of 8/25: I've spent most of the weekend trying to make the encoder step of the Pipeline work.  I can make encode/one-hot encoding functions work outside the Pipeline, but so far have found about 10 different ways it won't work in the Pipeline.  I'll work on this a bit more tomorrow but if I can't make it work for our data, will stick with the clunkier, pandas based code I wrote a few weeks ago to create dummy variables.  Less elegant but I know it works.
